File: algorithms/2_medium/M/01833_maximum_ice_cream_bars/solution.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        ranges::sort(costs);
        int count{ 0 };

        for (const auto& cost : costs) {
            if (cost <= coins) {
                coins -= cost;
                count += 1;
            } else {
                break;
            }
        }

        return count;
    }
};

File: algorithms/2_medium/M/01833_maximum_ice_cream_bars/solution.rs
impl Solution {
    pub fn max_ice_cream(mut costs: Vec<i32>, mut coins: i32) -> i32 {
        costs.sort_unstable();
        let mut count = 0_i32;

        for cost in costs.into_iter() {
            if cost <= coins {
                coins -= cost;
                count += 1;
            } else {
                break;
            }
        }

        count
    }
}
